youtube_dl/extractor/zingmp3.py

```diff
@@ -1,23 +1,52 @@
 # coding: utf-8
 from __future__ import unicode_literals
 
+import datetime
+import hashlib
+import hmac
+import re
+from urllib.parse import urlencode
+
 from .common import InfoExtractor
 from ..utils import (
-    ExtractorError,
     int_or_none,
+    try_get,
 )
 
 
 class ZingMp3BaseIE(InfoExtractor):
-    _VALID_URL_TMPL = r'https?://(?:mp3\.zing|zingmp3)\.vn/(?:%s)/[^/]+/(?P<id>\w+)\.html'
+    _VALID_URL_TMPL = r'https?://(?:mp3\.zing|zingmp3)\.vn/(?P<type>(?:%s))/[^/]+/(?P<id>\w+)(?:\.html|\?)'
     _GEO_COUNTRIES = ['VN']
+    _IS_UPDATE_COOKIES = False
+    _DOMAIN = 'https://zingmp3.vn'
+    _SLUG_API = {
+        'bai-hat': '/api/v2/page/get/song',
+        'embed': '/api/v2/page/get/song',
+        'video-clip': '/api/v2/page/get/video',
+        'playlist': '/api/v2/page/get/playlist',
+        'album': '/api/v2/page/get/playlist',
+        'lyric': '/api/v2/lyric/get/lyric',
+        'song_streaming': '/api/v2/song/get/streaming',
+    }
+
+    _TIMESTAMP = str(int(datetime.datetime.now().timestamp()))
+    _API_KEY = '88265e23d4284f25963e6eedac8fbfa3'
+    _SECRET_KEY = b'2aa2d1c561e809b267f3638c4a307aab'
+
+    def _extract_item(self, item, song_id, type_url, fatal):
+        item_id = item.get('encodeId') or song_id
+        title = item.get('title') or item.get('alias')
```

Comment:
`title` is required, so use the original pattern `item.get('title') or item['alias']`

Or if you want to make the ensuing crash report clearer, `dict_get(item, ('title', 'alias')) or item['title']`

Reply:
Yes, i updated